### ISSUE TITLE
Activate 'autorefresh' button in leaderboard view per default in case of live races

### ISSUE THREAD

**Issue by aksajhfduwafe** [aksajhfduwafe/bugzilla-to-github-issues-test#3749]
| Field | Value |
|---|---|
| **Reporter** | @fmittag |
| **Assignee** | Steffen Schäfer (`steffen.schaefer@pgt.de`) |
| **Product** | Sailing Race Analytics |
| **Component** | Frontend |
| **Version** | unspecified |
| **Hardware** | All |
| **OS** | All |
| **Severity** | enhancement |
| **Priority** | P3 |
| **CC (unmapped)** | Benjamin Barth (`benjamin.barth@trivadis.com`), Oleg Zheleznov (`Oleg_Zheleznov@epam.com`), Steffen Schäfer (`steffen.schaefer@pgt.de`) |
| **Created** | 2016-06-27T10:20:12Z |
| **Original Bug** | [Bug 3749](https://bugzilla.sapsailing.com/bugzilla/show_bug.cgi?id=3749) |

---

In the leaderboard view we have an indicator for live races highlighting some information with read color to make clear this leaderboard is a live leaderboard.
But the leaderboard autorefresh button is off per default, which is somehow inconsistent from my point of view.
I also made the observation that most people simply don't find/understand the autorefresh button and therefore never see our 'live leaderboard' feature.
Therefore I vote for the activation of the button per default in case of live races.

**Comment by aksajhfduwafe** [comment]
**@fmittag** commented:

The solution approach seems to be right, but as the LeaderboardUpdateListener is removed directly after the first call I wonder how we should deal with races which become live some time AFTER the leaderboard page has been opened by a user. Do we want the 'autorefresh' functionality automatically to start in such cases?

I think we should also reflect more that the page is updating it's content. The 'red' autorefresh button does not indicated this clearly enough from my point of view.

**Comment by aksajhfduwafe** [comment]
**@axeluhl** commented:

Please keep it simple. The decision whether to start with auto-refresh turned on should be made only once when the page is loaded. I don't think we should keep asking the server periodically whether any of the leaderboard races' live state has changed.

In case there is continued lack of satisfaction with the look and feel of the display of the auto-refresh button in its different states, please feel free to open a new bug. This bug here should remain focused on the start-up issue.

Frank, in your comment 4 (on this issue) you mentioned that the approach was correct for what was originally asked. Can this then be merged into master and resolved as fixed?

**Comment by aksajhfduwafe** [comment]
**Benjamin Barth (`benjamin.barth@trivadis.com`)** commented:

I've tested the lasted version on [bug3749](https://github.com/aksajhfduwafe/bugzilla-to-github-issues-test/tree/bug3749) with is commit 63a33ae19cfc8234f4879c03a12abc2c5ff0abdc.

The misbehaviors described in comment 14 (on this issue) are still present: "Currently this only works when accessing the leaderboard the first time. When switching the tab and go back to Leaderboard tab, autorefresh is not initially set." Also when accessing the regatta leaderboards tab on the series page the autorefresh seems not to be initialized, too.
